Dangerfile

```diff
@@ -22,13 +22,25 @@ changed_files = git.diff.stats[:files]
 excluded_changed_files = changed_files.select { |key| EXCLUSIONS.any? { |exclusion| key.include?(exclusion) } }
 filtered_changed_files = changed_files.reject { |key| EXCLUSIONS.any? { |exclusion| key.include?(exclusion) } }
 
+# fetch master for diff comparison
+`git fetch origin master`
+
+# get branch name
+branches = `git branch`
+current_branch = branches.match(/\*\s+([^\s]+)/m)[1]
+
 # ignores whitespace for the purpose of determining lines of code changed
-changes = `git diff master... -w --stat`.split("\n")
-lines_of_code = changes.sum(0) do |change|
-  if change == changes.last || EXCLUSIONS.any? { |exclusion| change.match(exclusion) }
-    0
-  else
-    change.match(/\|\s+(\d+)/)[1].to_i
+diff_changes = `git diff master..#{current_branch} -w --stat`.split("\n")
+
+if diff_changes.empty?
+  lines_of_code = filtered_changed_files.sum { |_file, changes| (changes[:insertions] + changes[:deletions]) }
+else
+  lines_of_code = diff_changes.sum(0) do |change|
+    if change == diff_changes.last || EXCLUSIONS.any? { |exclusion| change.match(exclusion) } || change.match(/\|\s+Bin/)
+      0
+    else
+      change.match(/\|\s+(\d+)/)[1].to_i
```

Comment:
What's the regex doing here?

Reply:
As it iterates through the diff list of files changed, it is checking to see if it is the last line of the diff (which isn't an actual part of the list of files changed) or an excluded file and if so, do not count it in the sum. Otherwise, grab the lines of code changed for that file, make it an integer, and add it to the sum. An example of what it spits out is below. So we would want it to grab the 3 and the 1 and add them to the sum but ignore the last line that is "2 files changed, 4 insertions(+)".

 Dangerfile       | 3 +++
 config/routes.rb | 1 +
 2 files changed, 4 insertions(+)

Reply:
Maybe add a description to the regex then. Like so
```suggestion
      changed_lines_count_regex = /\|\s+(\d+)/ # Captures the number in: | 123 ++++
      change.match(changed_lines_count_regex)[1].to_i
```

Same for the binary regex above.